File: backend/src/scripts/testUpload.js
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';
import { v2 as cloudinary } from 'cloudinary';
import dns from 'dns';

// Force Node to resolve IPv4 first (prevents timeouts on IPv6-unsupported networks)
if (typeof dns.setDefaultResultOrder === 'function') {
  dns.setDefaultResultOrder('ipv4first');
}

// Fix for ES modules __dirname equivalent
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load env variables
dotenv.config({ path: path.join(__dirname, '../../.env') });

// Configure Cloudinary
cloudinary.config({
  cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
  api_key: process.env.CLOUDINARY_API_KEY,
  api_secret: process.env.CLOUDINARY_API_SECRET
});

const imagePath = '/home/darkdevil404/.gemini/antigravity/brain/9b496ae7-87af-42d7-9294-34f397127ce0/test_design_1782226108423.png';
const folderName = process.env.CLOUDINARY_FOLDER || 'cf7a6f7a0c891c72efcc48041e47f299f8';

console.log('🔗 Uploading test image to Cloudinary...');
console.log('📂 Source file:', imagePath);
console.log('☁️ Cloud Name:', process.env.CLOUDINARY_CLOUD_NAME);
console.log('📁 Destination Folder:', folderName);

try {
  const result = await cloudinary.uploader.upload(imagePath, {
    folder: folderName
  });

  console.log('\n✅ Upload Successful!');
  console.log('═'.repeat(40));
  console.log('Asset ID:', result.asset_id);
  console.log('Public ID:', result.public_id);
  console.log('Secure URL:', result.secure_url);
  console.log('Folder:', result.folder);
  console.log('Format:', result.format);
  console.log('Dimensions:', `${result.width}x${result.height}`);
  console.log('═'.repeat(40));
} catch (error) {
  console.error('\n❌ Upload Failed!');
  console.error(error);
}

File: backend/src/scripts/test_upload.py
import os
import requests
import time
import hashlib

cloud_name = "c-e82cbded306108d674dd68715f5333"
api_key = "675432471277472"
api_secret = "EqAQ5F_aLmlXUtlhZ33S6FhltEQ"
folder = "cf7a6f7a0c891c72efcc48041e47f299f8"
file_path = "/home/darkdevil404/.gemini/antigravity/brain/9b496ae7-87af-42d7-9294-34f397127ce0/test_design_1782226108423.png"

url = f"https://api.cloudinary.com/v1_1/{cloud_name}/image/upload"

timestamp = str(int(time.time()))

# Parameters to sign (sorted alphabetically)
params_to_sign = f"folder={folder}&timestamp={timestamp}"

# Append API secret to string
string_to_hash = params_to_sign + api_secret

# Generate SHA1 signature
signature = hashlib.sha1(string_to_hash.encode('utf-8')).hexdigest()

files = {
    'file': open(file_path, 'rb')
}

data = {
    'api_key': api_key,
    'timestamp': timestamp,
    'folder': folder,
    'signature': signature
}

print(f"Sending request to: {url}")
try:
    response = requests.post(url, files=files, data=data, timeout=30)
    print(f"Status Code: {response.status_code}")
    print(response.json())
except Exception as e:
    print(f"Error occurred: {e}")
